### PR TITLE
feat: run against higher order k8s kinds

### LINKED ISSUE #27: Update policy capabilities-psp-policy to target high level resources capabilities-psp-policy
Update the policy from targeting Pod resource to target higher level objects like deployments. This is a better practice because we prevent the resources to be created instead of failing in the deployment phase when the pods are created.

### PR DESCRIPTION
**Pull request by KhaledEmaraDev**
## Description

Fix #27

## Test

To test this pull request, you can run the following commands:

### CODE REVIEW

**Review by jvanz — CHANGES_REQUESTED** [review]

**Review by jvanz — CHANGES_REQUESTED** [review]

File: src/validate.rs [hunk 1]
```diff
@@ -44,20 +48,33 @@ pub(crate) fn validate_added_caps(validation_req: &ValidationRequest<Settings>)
     Ok(())
 }
 
-fn get_caps(
-    queries: &[&str],
-    validation_req: &ValidationRequest<Settings>,
-) -> Result<HashSet<String>> {
-    let mut selector = jsonpath::selector_as::<HashSet<String>>(&validation_req.request.object);
-
+fn get_caps(pod_spec: &PodSpec) -> Result<HashSet<String>> {
```
[the remark this answers is not in the document]
Comment (jvanz) [reply src/validate.rs:51]: Yes, that was what I was trying to say. :)

[the remark this answers is not in the document]
Comment (KhaledEmaraDev) [reply src/validate.rs:51]: @jvanz Done. :)

**Review by viccuad — CHANGES_REQUESTED** [review]
LGTM (some linter errors, also missing commiting artifacthub-pkg.yml to history)

Yet, is there a specific reason why we don't validate ephemeral containers? I cannot recall one, they have a securityContext, hence we can set capabilities for them (and an attacker can use it for privilege escalation). I suppose it's because the policy predates their inclusion and the analogous PSP didn't do it either for the same reason.

I would take that line from the readme (or add the reason if there's any), and open an issue to check for caps on ephemeral containers.

**Review by viccuad — APPROVED** [review]
Approving, I'm ok merging as it is. Still needs linters fixes, please fix those.

Opened https://github.com/kubewarden/capabilities-psp-policy/issues/41 to also check ephemeral containers.

**Comment by jvanz** [comment]
> Approving, I'm ok merging as it is. Still needs linters fixes, please fix those.

@KhaledEmaraDev, please do it.

**Comment by KhaledEmaraDev** [comment]
@jvanz @viccuad Done.

**Review by jvanz — APPROVED** [review]

**Pull request merged**